cfgen/cfgen

```diff
@@ -17,10 +17,48 @@ import sys
 from typing import (Any, Callable, Dict, Iterable, Iterator, List, NamedTuple,
                     Optional, Set, Tuple)
 import yaml
+import logging
 
 
 __version__ = '0.14'
 
+LOG = logging.getLogger('cfgen')
+
+
+def repeat_on_cmd_timeout(f, max_retries=-1):
```

Comment:
I think we can use hax/hax/utils.py::repeat_if_fails() here. Pass `wait_seconds=15`.

Reply:
But it is for HAConsistency exception.

Reply:
>It is for HAConsistency exception.
you can do something like,
```
def repeat_if_fails(wait_seconds=5, max_retries=-1):
    """
    Ensures that the wrapped function gets re-invoked if
    HAConsistencyException gets raised. In other words, this wrapper
    makes the wrapped function repeatable.

    Parameters:

    wait_seconds - delay (in seconds) between the attempts. The delay
         applies after HAConsistencyException is raised.
    max_retries - how many attempts the wrapper will perform until finally
         re-raising the exception. -1 means 'repeat forever'.
    """
    def callable(f):
        @wraps(f)
        def wrapper(*args, **kwds):
            attempt_count = 0
            while (True):
                try:
                    return f(*args, **kwds)
                except HAConsistencyException as e:
                    attempt_count += 1
                    if max_retries >= 0 and attempt_count > max_retries:
                        LOG.warn(
                            'Function %s: Too many errors happened in a row '
                            '(max_retries = %d)', f.__name__, max_retries)
                        raise e
                    LOG.warn(f'Got HAConsistencyException: {e.message} while '
                             f'invoking function {f.__name__} '
                             f'(attempt {attempt_count}). The attempt will be '
                             f'repeated in {wait_seconds} seconds')
                    sleep(wait_seconds)
               except subprocess.TimeoutExpired as e:
                    ....
        return wrapper
```

You can also try, 
```
except Exception, e:
    if e.__class__ == <exception class>:
```
>The wrapper function inside needs the multiple positional arguments support.

Sorry didn't get this.

If we have functions doing the same things we should try and restrict it to a single version of them possibly, unless we have a very good reason to have multiple versions.

Reply:
```
The wrapper function inside needs the multiple positional arguments support.
```
This was an error I was getting while I was trying to do exactly same thing yesterday. So I had mentioned the reason of positional arguments.

But yes, I had thought about exactly same solution as well but again for that there was a problem I was facing that we unfortunately can not do this in the `cfgen ` file 
`from hax.util import repeat_if_fails'

I was also re-thinking on the same today again, and also had discussed this issue with @knekrasov . but I got to know from him that its not simple to import it in cfgen.

Reply:
Here is the explanation I got from @knekrasov today :
```
[6:25 PM] Konstantin Nekrasov
    Nowadays cfgen doesn't form a valid python package. So it stands  a bit 'outside'. And our scripts that invoke cfgen now don't make sure that hax' virtualenv is activated beforehand.
```

Reply:
Okay.